### PR TITLE
Feature/what it should have been

### PR DESCRIPTION
**Pull request by sb43**
removed diff_bams test, it was skipped for travis earlier, now removed altogether.
repo base is now develop.

### CODE REVIEW

File: git-hooks/pre-commit [hunk 1]
```diff
@@ -45,9 +48,20 @@ fi
...
 code=$?
 
-./run_tests.sh
+if [ "$code" != "0" ]
+	then
+		exit $code
+fi
+
+MYPACKAGE=archCompare
+#git diff --cached --name-only --diff-filter=ACM | xargs -I {} pycodestyle {}
```
Comment (keiranmraine) [inline git-hooks/pre-commit]: Why a comment, should this just be removed?

File: tests/test_check_input.py [hunk 2]
```diff
@@ -20,9 +20,9 @@ class TestClass():
     t_dirbamA=testdir+'testBamA'
     t_dirbamB=testdir+'testBamB'
     cwdpath=os.getcwd()
-    configdir = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'config/')
-    t_json=configdir+'fileTypes.json'
-   
+    configdir = os.path.join(os.path.dirname(os.path.realpath(__file__)), '../archCompare/config/')
+    t_json=configdir+'compareMethods.json'
```
Comment (keiranmraine) [inline tests/test_check_input.py]: Don't use the config dir stuff in a test.  A `json` files should exist in the `tests/data` tree for each specific thing they are testing (named appropriately).

File: tests/test_do_comparison.py [hunk 3]
```diff
@@ -20,72 +20,56 @@ class TestClass():
...
-
-
-    format_tar_dictB={
-                      'bam': [cwdpath+'/testB/samplea.bam', 'samplea.bam', '.bam'],
-                      'caveman_c.annot.vcf.gz': [cwdpath+'/testB/vcf_data/samplea.caveman_c.annot.vcf.gz', 'samplea.caveman_c.annot.vcf.gz', '.vcf.gz'],
-                      'bam.bai': [cwdpath+'/testB/samplea.bam.bai', 'samplea.bam.bai', '.bai'],
-                      'onlyinB.txt': [cwdpath+'/testB/onlyinB.txt', 'onlyinB.txt', '.txt']}
-
-
+    configdir = os.path.join(os.path.dirname(os.path.realpath(__file__)), '../archCompare/config/')
```
Comment (keiranmraine) [inline tests/test_do_comparison.py]: again should be a file in `tests/data`

File: tests/test_format_input.py [hunk 4]
```diff
@@ -20,56 +20,43 @@ class TestClass():
...
-    cmp_type='data'
-    file_dict={'samplea.caveman_c.annot.vcf.gz':[t_filea,'samplea.caveman_c.annot.vcf.gz','.vcf.gz']}
-
-    format_dir_bamdiffA={'samplea.bam': [t_dirbamA+'/samplea.bam', 'samplea.bam', '.bam']}
-    format_dir_bamdiffB={'samplea.bam': [t_dirbamB+'/samplea.bam', 'samplea.bam', '.bam']}
-    common_files_bamdiff=['samplea.bam']
-    bamdiff_result={'samplea.bam': ['compared', None]}
-
-
+    configdir = os.path.join(os.path.dirname(os.path.realpath(__file__)), '../archCompare/config/')
```
Comment (keiranmraine) [inline tests/test_format_input.py]: as for others

File: README.md [hunk 5]
```diff
@@ -34,7 +34,10 @@ the driver code.
 
 ### cgpCompare
 
-Takes the input archives, files, folders and does the comparison for matching file types based on tools defined in `archCompare/config/*.json` file.
+Takes the input archives, files, folders and does the comparison for matching file types based on tools defined in `archCompare/config/CompareMethods.json` file.
+
```
[the remark this answers is not in the document]
Comment (sb43) [reply README.md]: Added required paramter, provided example json file

File: README.md [hunk 6]
```diff
@@ -46,10 +49,9 @@ The output is a tab separated columns containing:
 
 * `File_a`  - compared file name  from first archive
 * `File_b`  - compared file name  from second archive
-* `Status`  - comparsion status [ compared, skipped ]
-* `SimilarityBy` - if files are compared and found similar it will have one of the value [ name, data or checksum ] otherwise 'differ', reason if files were skipped from comparison
+* `<reportsOn columns>` - if files are compared and found similar it will have one of the value [ Y=similar, N=different, <ReasonForNoComparison> ]
```
[the remark this answers is not in the document]
Comment (sb43) [reply README.md]: Modified to PASS, FAIL

**Pull request merged**